docs/adding-new-prowjob.md

```diff
@@ -0,0 +1,32 @@
+# Adding a new Prowjob
+
+To add a new Prowjob:
+1. Add the job configuration under the appropriate file under the [templater folder](../templater/jobs) based on job type. The current configuration struct looks like this:
+```Go
+type JobConfig struct {
+	HasReleaseBranches bool
+	JobName            string
+	RunIfChanged       string
+	Branches           []string
+	MaxConcurrency     int
+	CronExpression     string
+	Timeout            string
+	ImageBuild         bool
+	RuntimeImage       string
+	LocalRegistry      bool
+	ExtraRefs          []*ExtraRef
+	ServiceAccountName string
+	EnvVars            []*EnvVar
+	Commands           []string
+	Resources          *Resources
+	BuildkitResources  *Resources
+	VolumeMounts       []*VolumeMount
+	Volumes            []*Volume
+}
+```
+This struct is extensible to support more fields and fine-tuned configuration.
+
+2. Run `make prowjobs -C templater` to re-generate the `jobs/` folder along with the new Prowjob.
```

Comment:
Lets make the presubmit run this and either 1) generate the new files and check them into the PR or 2) fail the presubmit if the generated files do not match what is checked in.

Reply:
Done